### ISSUE TITLE
ParadoxicalCondition false positive

### ISSUE THREAD

**Issue by iluuu1994** [vimeo/psalm#1597]
```php
$protocol = !empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off'
	? 'https' 
	: 'http';
```

> ERROR: ParadoxicalCondition - 5:5 - Found a paradox when evaluating $_SERVER['HTTPS'] of type non-empty-mixed and trying to reconcile it with a empty assertion

https://psalm.dev/r/a7a9403b08

The code looks ok, any idea why this error is thrown?

**Comment by weirdan** [comment]
Looks like a Psalm bug to me. 

Possible workaround is to narrow the type of $_SERVER['HTTPS'] to string prior to checking `!== 'off'`: https://psalm.dev/r/05d8cdbdc7

**Comment by muglug** [comment]
You could also use 
```php
$protocol = ($_SERVER['HTTPS'] ?? 'off') !== 'off' ? 'https' : 'http';
```

**Comment by iluuu1994** [comment]
I'm working on a legacy code base, I'm aware that the code is not optimal. But it's correct nonetheless so I though I'd create an issue.

**Comment by muglug** [comment]
Also this seems to only be a problem in ternaries: https://psalm.dev/r/a0261a227c

**Comment by muglug** [comment]
@iluuu1994 no I love legacy codebase issues. They expose all Psalm’s bugs

**Comment by iluuu1994** [comment]
> no I love legacy codebase issues

At least somebody loves it 😅

Thanks for the fast fix!
